File: app/src/main/java/fi/kaupunkifillarit/analytics/FeedbackEvents.java
package fi.kaupunkifillarit.analytics;

import android.content.DialogInterface;

import com.google.android.gms.analytics.HitBuilders;

import java.util.HashMap;
import java.util.Map;

public abstract class FeedbackEvents {
    private static final String CATEGORY = "Feedback";
    private static final Map<Integer, String> BUTTONS_TO_LABELS;

    static {
        BUTTONS_TO_LABELS = new HashMap<>(3);
        BUTTONS_TO_LABELS.put(DialogInterface.BUTTON_POSITIVE, "Rate");
        BUTTONS_TO_LABELS.put(DialogInterface.BUTTON_NEUTRAL, "Later");
        BUTTONS_TO_LABELS.put(DialogInterface.BUTTON_NEGATIVE, "No thanks");
    }

    public static Map<String, String> showDialog() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_SHOW)
                .setValue(1)
                .build();
    }

    public static Map<String, String> buttonClick(int which) {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLICK)
                .setLabel(BUTTONS_TO_LABELS.get(which))
                .setValue(1)
                .build();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/analytics/FeedbackEvents.kt
package fi.kaupunkifillarit.analytics

import android.content.DialogInterface
import com.google.android.gms.analytics.HitBuilders

object FeedbackEvents {
    private const val CATEGORY = "Feedback"
    private val BUTTONS_TO_LABELS = mapOf(
            Pair(DialogInterface.BUTTON_POSITIVE, "Rate"),
            Pair(DialogInterface.BUTTON_NEUTRAL, "Later"),
            Pair(DialogInterface.BUTTON_NEGATIVE, "No thanks"))

    fun showDialog(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_SHOW)
                .setValue(1)
                .build()
    }

    fun buttonClick(which: Int): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLICK)
                .setLabel(BUTTONS_TO_LABELS[which])
                .setValue(1)
                .build()
    }
}
